### ISSUE TITLE
Before Suite running and failing occasionally on gci-gce-alpha-features job in master-blocking and release-1.11-blocking tests

### ISSUE THREAD

**Issue by AishSundar** [kubernetes/kubernetes#65192]
## Failing Jobs
* [sig-release-master-blocking#gci-gce-alpha-features](https://k8s-testgrid.appspot.com/sig-release-master-blocking#gci-gce-alpha-features&show-stale-tests=)
* [sig-release-1.11-blocking#gce-alpha-features-1.11](https://k8s-testgrid.appspot.com/sig-release-1.11-blocking#gce-alpha-features-1.11&show-stale-tests=)

## Failing Test
* [BeforeSuite](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-alpha-features/22718#beforesuite)

Whenever the BeforeSuite test runs (which is not deterministic from the dashboard), it fails and skips majority of other tests in the job. FWIW this has been the status of this job since release-1.9 and does not seem to be blocking a release. But it will be useful to get to the bottom of this flake and make sure there is no blocking Product issue here.

/kind bug
/priority failing-test
/priority critical-urgent
/assign @wenjiaswe
/milestone v1.11

/cc @jberkus @tpepper

**Comment by yguo0905** [comment]
From https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-cos-k8sbeta-alphafeatures/3364?log#log:

Looks like the node does not have enough resource to run the addon pods.

```
I0617 23:52:23.499] Jun 17 23:52:23.498: INFO: The status of Pod fluentd-gcp-v3.0.0-ll47q is Pending (Ready = false), waiting for it to be either Running (with Ready = true) or Failed
I0617 23:52:23.499] Jun 17 23:52:23.498: INFO: 28 / 33 pods in namespace 'kube-system' are running and ready (2 seconds elapsed)
I0617 23:52:23.499] Jun 17 23:52:23.498: INFO: expected 9 pod replicas in namespace 'kube-system', 9 are Running and Ready.
I0617 23:52:23.500] Jun 17 23:52:23.498: INFO: POD                       NODE  PHASE    GRACE  CONDITIONS
I0617 23:52:23.500] Jun 17 23:52:23.498: INFO: fluentd-gcp-v3.0.0-ll47q        Pending         [{PodScheduled False 0001-01-01 00:00:00 +0000 UTC 2018-06-17 23:52:21 +0000 UTC Unschedulable 0/4 nodes are available: 1 Insufficient cpu, 3 node(s) didn't match node selector.}]
```

**Comment by AishSundar** [comment]
@yguo0905 it looks like a test setup issue which has prolonged for a while and no product issues of concern. Correct me otherwise.

From offline discussion, looks like BeforeSuite always runs and does NOT report back if it passes. Hence the test seems to fail intermittently due to resource issues. as per @krzyzacy all the cluster has the same shape (3 nodes,  n1-standard-2), so we are not very only 1.11 alpha-feature suite is flaky. We are trying to find some real test owners from sig-gcp from last week, but no response.

**Comment by AishSundar** [comment]
FWIW, the alpha-features job and specifically the BeforeSuite started failing this way only since 3/8/2018 and has been during 1.10 RC cut timeframe as well

https://storage.googleapis.com/k8s-gubernator/triage/index.html?date=2018-03-15&job=e2e-gci-gce-alpha-features&test=BeforeSuite

So some feature introduced in alpha / test infra changes around that time is causing resource constraints. @krzyzacy @idvoretskyi is there any way we can track which feature got into alpha around 3/8/2018 timeframe and track down owners from there?

**Comment by krzyzacy** [comment]
hummm we can... compare junit entries and see what features are newly added?

**Comment by bsalamat** [comment]
I cannot completely rule out that this is related to "schedule DS pods by default scheduler", but given that the test has been flaky in older releases, I doubt if the new alpha feature is causing it.
It looks like the node does not have enough resources to run the addon. Whether the scheduler schedules these addons or DaemonSet Controller does it, when a node is out of resources the daemons cannot be scheduled.

**Comment by AishSundar** [comment]
"but given that the test has been flaky in older releases, I doubt if the new alpha feature is causing it"
@bsalamat mining for [historic health of this test from bigquery](https://storage.googleapis.com/k8s-gubernator/triage/index.html?date=2018-03-15&job=e2e-gci-gce-alpha-features&test=BeforeSuite), I see the Beforesuite started failing this way only since 3/8/2018

**Comment by k82cn** [comment]
> I see the Beforesuite started failing this way only since 3/8/2018

@AishSundar / @dims , for the time window, it should not relate to it; the major PR (https://github.com/kubernetes/kubernetes/pull/63223) was merged at early this month (Jun 2018) :)

But if no one working on this, I'd like to help to unblock release :)

**Comment by dims** [comment]
@k82cn one of the first PRs for the feature (https://github.com/kubernetes/kubernetes/pull/59862) was merged on 3/11/2018 :)

**Comment by k82cn** [comment]
wow, that PR was so powerful, making the CI failed (3/8) before it merged (3/11) :)

**Comment by k82cn** [comment]
No clue right now, but something interesting:

According to the following log, there's no sufficient cpu for `fluentd-gcp-v3.0.0-kwbqj`; but `fluentd-gcp-v3.0.0-kwbqj` is `BestEffort` according to its [yaml](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml) : in scheduler, `PodFitResource` predicate will not check resource for `BestEffort` Pods.

```
I0620 20:48:17.567] Jun 20 20:48:17.565: INFO: fluentd-gcp-v3.0.0-kwbqj                                            Pending         [{PodScheduled False 0001-01-01 00:00:00 +0000 UTC 2018-06-20 20:48:16 +0000 UTC Unschedulable 0/4 nodes are available: 1 Insufficient cpu, 3 node(s) didn't match node selector.}]
```

**Comment by dims** [comment]
LOL. look at the graph in https://storage.googleapis.com/k8s-gubernator/triage/index.html?date=2018-03-15&job=e2e-gci-gce-alpha-features&test=BeforeSuite there are 2 flakes before 3/8 and a bunch after 3/11 :)

**Comment by dims** [comment]
@BenTheElder is it possible to bump up the available cpu for nodes in this job?

**Comment by k82cn** [comment]
Still not sure why fluentd-gcp-v3.0.0-kwbqj asked cpu resource, and the cpu gap is 37 (MillValue); @piosz @x13n . BTW, also check some other logs of flaky test, the cpu gap is always 37 (MillValue); maybe we can try to enlarge the allocatable of node.

Here's another issue: in scheduler, it found `fluentd-gcp-v3.0.0-kwbqj` can not be scheduled because of insufficient cpu and trigger
preemption (node-critical > cluster-critical); but in kubelet, it'll not preempt any 'critical' pods, refer to the following logs for
the detail. This issue did not contribute to the flaky test.


scheduler log:

```
I0620 20:48:16.832423       1 factory.go:1416] Updating pod condition for kube-system/fluentd-gcp-v3.0.0-kwbqj to (PodScheduled==False)
I0620 20:48:16.871162       1 generic_scheduler.go:1002] Node bootstrap-e2e-master is a potential node for preemption.
I0620 20:48:16.900337       1 factory.go:1181] About to try and schedule pod fluentd-gcp-v3.0.0-kwbqj
I0620 20:48:16.900385       1 scheduler.go:447] Attempting to schedule pod: kube-system/fluentd-gcp-v3.0.0-kwbqj
I0620 20:48:16.900799       1 scheduler.go:194] Failed to schedule pod: kube-system/fluentd-gcp-v3.0.0-kwbqj
I0620 20:48:16.900835       1 factory.go:1303] Unable to schedule kube-system fluentd-gcp-v3.0.0-kwbqj: no fit: 0/4 nodes are available: 1 Insufficient cpu, 3 node(s) didn't match node selector.; waiting
I0620 20:48:16.900887       1 factory.go:1416] Updating pod condition for kube-system/fluentd-gcp-v3.0.0-kwbqj to (PodScheduled==False)
I0620 20:48:17.761558       1 factory.go:1181] About to try and schedule pod fluentd-gcp-v3.0.0-kwbqj
I0620 20:48:17.761618       1 scheduler.go:447] Attempting to schedule pod: kube-system/fluentd-gcp-v3.0.0-kwbqj
I0620 20:48:17.769339       1 scheduler_binder.go:194] AssumePodVolumes for pod "kube-system/fluentd-gcp-v3.0.0-kwbqj", node "bootstrap-e2e-master"
I0620 20:48:17.769422       1 scheduler_binder.go:197] AssumePodVolumes for pod "kube-system/fluentd-gcp-v3.0.0-kwbqj", node "bootstrap-e2e-master": all PVCs bound and nothing to do
I0620 20:48:17.769507       1 factory.go:1407] Attempting to bind fluentd-gcp-v3.0.0-kwbqj to bootstrap-e2e-master
```

kubelet log:

```
W0620 20:48:17.949156    1506 predicate.go:107] Failed to admit pod fluentd-gcp-v3.0.0-kwbqj_kube-system(41eec71a-74cb-11e8-854a-42010a800002) - Unexpected error while attempting to recover from admission failure: preemption: error finding a set of pods to preempt: no set of running pods found to reclaim resources: [(res: cpu, q: 37), ]
```

**Comment by BenTheElder** [comment]
@dims we can set the nodes to any size but I think the default is
n1-standard-4.. I'd be a little concerned that anything besides maybe
hollow kubelets isn't fitting on default nodes 😱
Will follow up.

**Comment by BenTheElder** [comment]
Looks like it's actually `n1-standard-2` nodes per [the logs](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-alpha-features/22807/?log#log), FWIW.

**Comment by k82cn** [comment]
@BenTheElder , is there anything I need to follow up ?

**Comment by BenTheElder** [comment]
I'm not sure, I'd be somewhat hesitant to just throw bigger nodes at it and hope that works. If we have to keep using bigger nodes to run the basic tests, what are end users doing? :|

**Comment by k82cn** [comment]
I'd like to see what happened after #65372 merged, refer to [comments](https://github.com/kubernetes/kubernetes/issues/65372#issuecomment-400873429) for detail

**Comment by k82cn** [comment]
#65372 was merged, let's see what happens to the CI :)

**Comment by jberkus** [comment]
This is still failing in 1.11, but not on master.

As such, demoting priority, but it still needs to be fixed for the sake of 1.11 users.

/priority important-longterm
